clinicaltrials.gov.js

```diff
@@ -0,0 +1,273 @@
+				creators.push({
+					firstName: splitResponsiblePartyInvestigator[0],
+					lastName: splitResponsiblePartyInvestigator[1],
+					creatorType: "author"
+				});
+			}
+		}
+
+		if (study.ProtocolSection.SponsorCollaboratorsModule.hasOwnProperty("LeadSponsor")) {
+			sponsor = study.ProtocolSection.SponsorCollaboratorsModule.LeadSponsor.LeadSponsorName;
+			let sponsorCreatorType;
+			if (creators.length == 0) {
+				sponsorCreatorType = "author";
+			}
+			else {
+				sponsorCreatorType = "contributor";
+			}
+			creators.push({
+				firstName: sponsor,
+				creatorType: sponsorCreatorType
+			});
+		}
+		
+		if (study.ProtocolSection.SponsorCollaboratorsModule.hasOwnProperty("CollaboratorList")) {
+			const collaboratorList = study.ProtocolSection.SponsorCollaboratorsModule.CollaboratorList.Collaborator;
+			collaboratorList.forEach((collaborator) => {
+				collaborators.push(
+					{
+						firstName: collaborator.CollaboratorName,
+						creatorType: "contributor"
+					}
+				);
+			});
+			collaborators.forEach((collaborator) => {
+				creators.push(collaborator);
+			});
+		}
+
+		item.creators = creators;
+		// Done get the creator info
+
+		item.date = study.ProtocolSection.StatusModule.LastUpdateSubmitDate;
+		item.accessDate = dateTimeToDateString(data.FullStudiesResponse.DataVrs);
+		item.institution = "clinicaltrials.gov";
+		item.reportNumber = clinicalTrialID;
+		item.shortTitle = study.ProtocolSection.IdentificationModule.BriefTitle;
+		item.abstractNote = study.ProtocolSection.DescriptionModule.BriefSummary;
+		item.url = "https://clinicaltrials.gov/ct2/show/" + clinicalTrialID;
+		item.reportType = "Clinical trial registration";
+		item.extra = `submitted: ${study.ProtocolSection.StatusModule.StudyFirstSubmitDate}`;
+		item.complete();
+	});
+}
+
+
+/** BEGIN TEST CASES **/
+var testCases = [ 
+	{
+		"type": "web",
+		"url": "https://clinicaltrials.gov/ct2/show/NCT04292899",
+		"items": [
+			{
+				"itemType": "report",
+				"title": "A Phase 3 Randomized Study to Evaluate the Safety and Antiviral Activity of Remdesivir (GS-5734™) in Participants With Severe COVID-19",
+				"abstractNote": "The primary objective of this study is to evaluate the efficacy of 2 remdesivir (RDV) regimens with respect to the normalization of temperature and oxygen saturation through Day 14 in participants with severe coronavirus disease (COVID-19).",
+				"creators": [
+					{
+						"firstName": "Gilead Sciences",
+						"creatorType": "author"
+					}
+				],
+				"date": "April 1, 2020",
+				"accessDate": "2020-04-01",
+				"libraryCatalog": "clinicaltrials.gov",
+				"shortTitle": "Study to Evaluate the Safety and Antiviral Activity of Remdesivir (GS-5734™) in Participants With Severe Coronavirus Disease (COVID-19)",
+				"url": "https://clinicaltrials.gov/ct2/show/NCT04292899",
+				"institution": "clinicaltrials.gov",
+         		"reportNumber": "NCT04292899",
```

Comment:
are the test generated with scaffold? It's odd that the indenting is off for report details. Let's fix this

Reply:
I fixed the indenting issue. I generated the tests myself without using scaffold; I think I tried using it but had some issues getting it running if I remember correctly.

Reply:
Just to clarify, Scaffold is now just Tools → Developer → Translator Editor in Zotero, not a separate plugin. The test needs to be generated by Scaffold to be valid. (If you generated it by hand, how did you test it?)

If you're having trouble with something in Scaffold, let us know.

Reply:
Thanks for clarifying that @dstillman. That's how I tested it (and now that I'm looking at it again I can see that it says "Scaffold" at the top of the window... ). That was very helpful for developing the translator and how I tested it (I also used a simple node.js script to test the parsing while developing it so that I could get more instantaneous feedback)